File: mobile_app/app/src/main/java/com/example/licenta_mobile/ReservationExtensionActivity.java
package com.example.licenta_mobile;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ListView;

import com.example.licenta_mobile.adapter.PendingReservationAdapter;
import com.example.licenta_mobile.dto.UnconfirmedReservationDto;
import com.example.licenta_mobile.model.UserData;
import com.example.licenta_mobile.rest.ReservationService;
import com.example.licenta_mobile.rest.RestClient;
import com.example.licenta_mobile.security.Token;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ReservationExtensionActivity extends AppCompatActivity {

    private ReservationService reservationService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reservations);
        reservationService = RestClient.getClient().create(ReservationService.class);
        handleUnconfirmedReservations();
    }

    @Override
    public void onBackPressed() {
        Intent intent = new Intent(this, ParkingActivity.class);
        startActivity(intent);
    }

    private void handleUnconfirmedReservations(){
        int userId = UserData.getUserId();
        Call<List<UnconfirmedReservationDto>> call = reservationService.getUnoccupiedPlaces("Bearer " + Token.getJwtToken(), userId);
        call.enqueue(new Callback<List<UnconfirmedReservationDto>>() {

            @Override
            public void onResponse(Call<List<UnconfirmedReservationDto>> call, Response<List<UnconfirmedReservationDto>> response) {
                if (response.isSuccessful()) {
                    List<UnconfirmedReservationDto> unconfirmedReservations = response.body();
                    buildView(unconfirmedReservations);
                }
            }

            @Override
            public void onFailure(Call<List<UnconfirmedReservationDto>> call, Throwable t) {
                System.out.println(t.getMessage());
                call.cancel();
            }
        });
    }

    private void buildView(List<UnconfirmedReservationDto> pendingReservations){
        NotificationHandler notificationHandler = new NotificationHandler(this);
        PendingReservationAdapter adapter = new PendingReservationAdapter(pendingReservations, this, notificationHandler, this);
        ListView lView = findViewById(R.id.pendingReservationList);
        lView.setAdapter(adapter);
    }


}

File: mobile_app/app/src/main/java/com/example/licenta_mobile/ReservationExtensionActivity.kt
package com.example.licenta_mobile

import android.content.Intent
import android.os.Bundle
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.licenta_mobile.adapter.PendingReservationAdapter
import com.example.licenta_mobile.dto.ReservationDto
import com.example.licenta_mobile.model.UserData.userId
import com.example.licenta_mobile.rest.ReservationService
import com.example.licenta_mobile.rest.RestClient.client
import com.example.licenta_mobile.security.Token
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ReservationExtensionActivity : AppCompatActivity() {
    private var reservationService: ReservationService? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reservations)
        reservationService = client!!.create(ReservationService::class.java)
        val viewTitle = findViewById<TextView>(R.id.reservationTitle)
        viewTitle.text = getString(R.string.pendingReservations)
        handleUnconfirmedReservations()
    }

    override fun onBackPressed() {
        val intent = Intent(this, ParkingActivity::class.java)
        startActivity(intent)
    }

    private fun handleUnconfirmedReservations() {
        val userId = userId!!
        val call = reservationService!!.getUnoccupiedPlaces("Bearer " + Token.jwtToken, userId)
        call.enqueue(object : Callback<MutableList<ReservationDto>?> {
            override fun onResponse(call: Call<MutableList<ReservationDto>?>, response: Response<MutableList<ReservationDto>?>) {
                if (response.isSuccessful) {
                    val unconfirmedReservations = response.body()
                    buildView(unconfirmedReservations)
                }
            }

            override fun onFailure(call: Call<MutableList<ReservationDto>?>, t: Throwable) {
                println(t.message)
                call.cancel()
            }
        })
    }

    private fun buildView(pendingReservations: MutableList<ReservationDto>?) {
        val notificationHandler = NotificationHandler(this)
        val adapter = PendingReservationAdapter(pendingReservations, notificationHandler, this)
        val lView = findViewById<ListView>(R.id.pendingReservationList)
        lView.adapter = adapter
    }
}